dali/test/operator_adapter.h

```diff
@@ -0,0 +1,23 @@
+#ifndef DALI_OPERATOR_ADAPTER_H
+#define DALI_OPERATOR_ADAPTER_H
+
+namespace dali {
+
+namespace testing {
+
+class OperatorAdapter {
+public:
+  enum Backend {
+    CPU, GPU,
+  };
+
+private:
+  std::string operator_name_;
+  Backend input_backend_, output_backend_;
```

Comment:
Wrong assumption: we already have operators which change backend (MixedBackend does extactly that?).

Reply:
`MixedBackend` has an input CPU backend and output GPU backend. What I tried to do here is to make some generalization, so that every inputbackend-outputbackend permutation will be possible.